File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.py
import sys

input = sys.stdin.readline
t = int(input())

for _ in range(t):
  n = int(input())
  clothes = {}

  for _ in range(n):
    name, category = input().strip().split()
    if category not in clothes:
      clothes[category] = []
    clothes[category].append(name)

  count = 1
  for c in clothes.values():
    count *= (len(c) + 1)
  print(count - 1)

File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());

        for (int tc = 0; tc < t; tc++){
            int n = Integer.parseInt(br.readLine());
            Map<String, List<String>> clothes = new HashMap<>();

            for (int i = 0; i < n; i++){
                StringTokenizer st = new StringTokenizer(br.readLine());
                String name = st.nextToken();
                String category = st.nextToken();

                if (!clothes.containsKey(category)){
                    clothes.put(category, new ArrayList<>());
                }
                clothes.get(category).add(name);
            }

            int count = 1;
            for (List<String> c : clothes.values()) {
                count *= (c.size() + 1);
            }
            System.out.println(count - 1);
        }
    }
}